src/System.Collections.Sequences/System/Collections/Sequences/Position.cs

```diff
@@ -16,20 +16,19 @@ namespace System.Collections.Sequences
 
         public static implicit operator Position(int index) => new Position(index, null);
```

Comment:
Do we need all these operators, `Get<T>`, and static `Create` method? 
Can we just expose item and index and add a single public constructor that takes them?

Reply:
Also in all of the usages of `Get<T>` you never check that item is not null. Do we need to add some kind of validation that position was of an expected type?

Reply:
That's how it was originally, but you guys wanted to make it harder for people to get access to the object. Get/Create allow you to store an internal type in object to make it impossible to retrieve by third party code.

Reply:
You can always do `Get<object>` and get around it. And `ReadOnlyBuffer` stores publicly available types there. I don't think we can protect it from retrieving the item.

Reply:
Yes, but I wanted to change the check to only return the object if typeof(T) == _item.GetType().
I was planing to do it in a subsequent PR.